### PR TITLE
RF_POM | menPage.spec | TC 04.3.4_03 <Men page> Category block contains sub-categories: Tops and Bottoms which are links in blue text

### PR DESCRIPTION
**Pull request by yauheniumetski**
https://trello.com/c/LbpsgWVN/611-rf-homepagespec-tc-043403-men-page-category-block-contains-sub-categories-tops-and-bottoms-which-are-links-in-blue-text

### CODE REVIEW

**Comment by yauheniumetski** [comment]
[RF_POM | menPage.spec | TC 04.3.4_03 <Men page> Category block contains sub-categories: Tops and Bottoms which are links in blue text](https://trello.com/c/LbpsgWVN/611-rfpom-menpagespec-tc-043403-men-page-category-block-contains-sub-categories-tops-and-bottoms-which-are-links-in-blue-text)

**Comment by yauheniumetski** [comment]
The first test was written without a test.describe(). I've just moved 1'st test into describe with my own

File: tests/test_page_objects/menPage.spec.js [hunk 1]
```diff
@@ -1,23 +1,46 @@
...
     COMPARE_PRODUCTS_TEXT,
     MY_WISH_LIST_TEXT
 } from "../../helpers/testMenData.js";
 import MenPage from "../../page_objects/menPage";
+import ConsentDataPopUp from "../../page_objects/consentDataPopUp.js";
 
-test('Menu/Men available to click, see clothes only for men', async ({ page }) => {
-    const homePage = new HomePage(page);
-    const menPage = new MenPage(page);
+test.describe('menPage.spec', () => {
```
Comment (S-Lanchik) [inline tests/test_page_objects/menPage.spec.js:13]: you should also create beforeEach with homepage.open()

File: tests/test_page_objects/menPage.spec.js [hunk 2]
```diff
@@ -1,23 +1,46 @@
...
+    test('Category block contains sub-categories: Tops and Bottoms which are links in blue text', async ({ page }) => {
+        const homePage = new HomePage(page);
+        const menPage = new MenPage(page);
+        const consentDataPopUp = new ConsentDataPopUp(page);
+
+        await homePage.open();
+        if (await consentDataPopUp.locators.getConsentDataPopUp().isVisible()) {
+            await consentDataPopUp.clickConsentDataButton();
+        };
+        await expect(page).toHaveURL(BASE_URL);
```
Comment (S-Lanchik) [inline tests/test_page_objects/menPage.spec.js:36]: Those lines are not necessary

File: tests/test_page_objects/menPage.spec.js [hunk 3]
```diff
@@ -1,23 +1,46 @@
...
+
+        await homePage.clickMenLink();
+        await expect(page).toHaveURL(BASE_URL + MEN_PAGE_END_POINT);
+        await expect(menPage.locators.getCategoryBlock()).toBeVisible();
+        await expect(menPage.locators.getCategoryBlock()).toHaveText(MEN_PAGE_CATEGORY_HEADER);
+
+        await expect(menPage.locators.getTopsSubCategoryLink()).toHaveCSS('color', MEN_PAGE_TOPS_SUB_CATEGORY_LINK_COLOR);
+        await expect(menPage.locators.getBottomsSubCategoryLink()).toHaveCSS('color', MEN_PAGE_BOTTOMS_SUB_CATEGORY_LINK_COLOR);
+    });
+});
```
Comment (S-Lanchik) [inline tests/test_page_objects/menPage.spec.js:46]: Your test is checking color of the links, so these two assertions are enough for this test

File: page_objects/consentDataPopUp.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        getConsentDataButton: () => this.page.getByRole('button', { name: 'Consent' })
+    };
+
+    async clickConsentDataButton() {
+        await this.locators.getConsentDataButton().click();
+
+        return new HomePage(this.page);
+    };
+}
+export default ConsentDataPopUp;
```
Comment (S-Lanchik) [inline page_objects/consentDataPopUp.js:19]: Usually this popup window only is on local machine. Try push your code without it.

File: tests/test_page_objects/menPage.spec.js (hunk 1 above)
Comment (yauheniumetski) [reply tests/test_page_objects/menPage.spec.js:12]: Fixed, thanks!

File: tests/test_page_objects/menPage.spec.js (hunk 2 above)
Comment (yauheniumetski) [reply tests/test_page_objects/menPage.spec.js]: Fixed, thanks!

**Review by S-Lanchik — APPROVED** [review]

**Review by SvetlanaMag — APPROVED** [review]

**Pull request merged**